src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2905,18 +2903,18 @@ def transform(dataset):
         ]),
         Bool('sparse'),
         Bool('force_size'),
-        Inheritable(Str('comments')),
-        Inheritable(Str('sync', enum=['STANDARD', 'ALWAYS', 'DISABLED'])),
-        Inheritable(Str('compression', enum=ZFS_COMPRESSION_ALGORITHM_CHOICES)),
+        Inheritable(Str('comments'), default='INHERIT'),
+        Inheritable(Str('sync', enum=['STANDARD', 'ALWAYS', 'DISABLED']), default='INHERIT'),
+        Inheritable(Str('compression', enum=ZFS_COMPRESSION_ALGORITHM_CHOICES), default='INHERIT'),
         Inheritable(Str('atime', enum=['ON', 'OFF'])),
-        Inheritable(Str('exec', enum=['ON', 'OFF'])),
-        Inheritable(Str('managedby', empty=False)),
+        Inheritable(Str('exec', enum=['ON', 'OFF']), default='INHERIT'),
+        Inheritable(Str('managedby', empty=False), default='INHERIT'),
         Int('quota', null=True, validators=[Or(Range(min=1024**3), Exact(0))]),
-        Inheritable(Int('quota_warning', validators=[Range(0, 100)])),
-        Inheritable(Int('quota_critical', validators=[Range(0, 100)])),
+        Inheritable(Int('quota_warning', validators=[Range(0, 100)]), default='INHERIT'),
+        Inheritable(Int('quota_critical', validators=[Range(0, 100)]), default='INHERIT'),
         Int('refquota', null=True, validators=[Or(Range(min=1024**3), Exact(0))]),
-        Inheritable(Int('refquota_warning', validators=[Range(0, 100)])),
-        Inheritable(Int('refquota_critical', validators=[Range(0, 100)])),
+        Inheritable(Int('refquota_warning', validators=[Range(0, 100)]), default='INHERIT'),
+        Inheritable(Int('refquota_critical', validators=[Range(0, 100)]), default='INHERIT'),
         Int('reservation'),
         Int('refreservation'),
         Inheritable(Int('special_small_block_size')),
```

Comment:
Don't you also need `default` here?

Reply:
No, this property is not valid for volumes

Reply:
So it works without `default`? Why do we need it then?

Reply:
`INHERIT` is a no-op, zfs create api will inherit if it's not specified explicitly. It's just been defined to be explicit rather then implicit.